### PR TITLE
rename Lasso filtering

### PR DESCRIPTION
**Pull request by riccardocadei**
rename `select_causal_rules()` function in `lasso_rules_filter()`

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 ### Changed
 
+* rename `select_causal_rules()` function in `lasso_rules_filter()` and
```
Comment (Naeemkh) [inline NEWS.md:5]: Please focus more on the current state of the code. For example, in this case, you can say: `select_causal_rules()` is now `lasso_rules_filter().`

File: NEWS.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 ### Changed
 
+* rename `select_causal_rules()` function in `lasso_rules_filter()` and 
+update its definition
+* add parameter `replace` to set replacement in bootstrapping for rules generation
```
Comment (Naeemkh) [inline NEWS.md:7]: In this one: rules generation now accepts `replace` parameter to set replacement in bootstrapping.

File: R/discard_correlated_rules.R [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  # Identify correlated rules
+  nrules <- length(rules_list)
+  ind <- 1:nrules
+  C <- stats::cor(rules_matrix)
+  elim <- c()
+  for(i in 1:(nrules - 1)) {
+    elim <- c(elim, which(round(abs(C[i, (i + 1):nrules]), digits = 4)
+                          >= t)
+              + i)
```
Comment (Naeemkh) [inline R/discard_correlated_rules.R:28]: Please reorganize this line. It is hard to read.

File: R/generate_causal_rules.R [hunk 4]
```diff
@@ -37,39 +38,53 @@ generate_causal_rules <- function(X, ite_std, method_params, hyper_params) {
 
   # 2.1 Prune irrelevant variable-value pair from a rule condition
   logger::log_info("Pruning ...")
-  rules <- prune_rules(rules,
+  rules_list <- prune_rules(rules,
                        X,
```
Comment (Naeemkh) [inline R/generate_causal_rules.R:42]: Please fix the indentation.

File: R/generate_rules.R [hunk 5]
```diff
@@ -36,20 +37,20 @@ generate_rules <- function(X, ite_std, ntrees_rf, ntrees_gbm, node_size,
...
-                                                        replace = FALSE,
+                                                        replace = replace,
                                                         ntree = 1,
                                                         maxnodes = mn,
-                                                        nodesize = node_size))
+                                                        nodesize = node_size)
   for(i in 2:ntrees_rf) {
     mn <- 2 + floor(stats::rexp(1, 1 / (max_nodes - 2)))
     set.seed(seed_2[i])
     model1_RF <- suppressWarnings(
```
Comment (Naeemkh) [inline R/generate_rules.R:49]: Let's keep consistency and remove the suppressWarnings for all randomforest function calls.

File: R/generate_rules_matrix.R [hunk 6]
```diff
@@ -6,70 +6,48 @@
...
+#' Standardize Rules Matrix
+#'
+#' @description
+#' Standardize the matrix of causal rules given a list.
+#'
+#' @param rules_matrix The rules matrix.
+#'
+#' @return
+#' Standardized rules matrix
+#'
```
Comment (Naeemkh) [inline R/generate_rules_matrix.R:39]: If this is an internal function, please add `@keywords internal`.

File: tests/testthat/test-interpret_select_rules.R [hunk 7]
```diff
@@ -41,19 +42,16 @@ test_that("Rules Interpreted Correctly", {
...
-  rules_matrix <- rules_all[["rules_matrix"]]
-  rules_matrix_std <- rules_all[["rules_matrix_std"]]
-  rules_list_dis <- rules_all[["rules_list"]]
+  rules_matrix <- generate_rules_matrix(X, rules_list)
+  rules_matrix_std <- standardize_rules_matrix(rules_matrix)
 
   # Step 5: Select important rules
-  select_rules <- as.character(select_causal_rules(rules_matrix_std, rules_list_dis,
+  select_rules <- as.character(lasso_rules_filter(rules_matrix_std, rules_list,
                                                    ite_std, q, stability_selection, pfer_val))
```
Comment (Naeemkh) [inline tests/testthat/test-interpret_select_rules.R:55]: More than 80 char.

**Review by Naeemkh — CHANGES_REQUESTED** [review]

File: R/generate_rules_matrix.R (hunk 6 above)
Comment (riccardocadei) [reply R/generate_rules_matrix.R:39]: what's the difference between internal and external?

Comment (Naeemkh) [reply R/generate_rules_matrix.R:41]: We `@keywords internal` or `export`. Those functions that we do not want to share publicly (on CRAN) should be internal. Similar to the private method in other languages.

Comment (riccardocadei) [reply R/generate_rules_matrix.R:41]: No, actually this function should be public. If I not set anything is it public by default?

Comment (riccardocadei) [reply R/generate_rules_matrix.R:40]: just fixed

**Pull request merged**